graphql-server/pkg/queries/proposal.go

```diff
@@ -121,3 +122,12 @@ func (q *ProposalQuery) QueryProposalTallyResults(ids []int) ([]*models.Proposal
 
 	return result, nil
 }
+
+func (q *ProposalQuery) QueryProposalByIDs(ids []string) ([]*models.Proposal, error) {
+	proposals := make([]*models.Proposal, len(ids))
+	err := q.NewQuery().Where("id IN (?)", bun.In(ids)).Scan(q.ctx, &proposals)
```

Comment:
Since we are using `IN` the order can be different than the input, this would cause the dataloader to return wrong results I think. We should retain the order using a map, ref `QueryBlocksByHashes`